src/ints.jl

```diff
@@ -111,14 +117,16 @@ end
     end
     b = peekbyte(source, pos)
     # parse rest of number
-    digits = Int64(0)
-    x, code, pos = parsedigits(Number, source, pos, len, b, code, opts, digits, neg, startpos)
+    _, code, pos = parsedigits(Number, source, pos, len, b, code, OPTIONS, Int64(0), neg, startpos, f)
     if (x === Inf || x === -Inf) && !specialvalue(code)
```

Comment:
`x` doesn't seem to exist when we get to the if statement

Reply:
IIUC, ideally, we'd know that we got an `Inf` here so we could try parsing as BigFloat

Reply:
Yeah, this is the remaining wrinkle; I think we can maybe use our `code` here w/ a special value to know if Inf was set. Let me play around with it in a bit.